src/run.rs

```diff
@@ -303,6 +303,78 @@ fn image_ids_in_use() -> io::Result<HashSet<String>> {
     Ok(image_ids)
 }
 
+// Find docker root directory.
+fn docker_root_dir() -> io::Result<String> {
+    // Query Docker for the root directory.
+    let output = Command::new("docker")
+        .args(&["info", "--format", "{{.DockerRootDir}}"])
+        .stderr(Stdio::inherit())
+        .output()?;
+
+    // Ensure the command succeeded.
+    if !output.status.success() {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Unable to determine the Docker root directory.",
+        ));
+    }
+
+    // Find the relevant line of output.
+    String::from_utf8(output.stdout)
+        .map_err(|error| io::Error::new(io::ErrorKind::Other, error))
+        .map(|s| s.trim().to_string())
+}
+
+// Find size of filesystem on which docker root directory is stored.
+fn docker_root_dir_filesystem_size() -> io::Result<Byte> {
+    let root_dir = docker_root_dir()?;
+
+    let output = Command::new("df")
+        .args(&["-B", "1", &root_dir])
+        .stderr(Stdio::inherit())
+        .output()?;
```

Comment:
This command would fail on Windows, wouldn't it? I'm not too thrilled about this dependency on `df`—is there any way to avoid it?

Reply:
I found this: https://crates.io/crates/dusage which seems to provide cross-platform support for getting disk sizes. It is missing the functionality of directly resolving which device the docker-directory is on, but i guess a longest-prefix match of docker-directory and mountpoint would be sufficient?

Reply:
I have integrated sysinfo, and the solution looks much more cross-platform capable now! I am however not able to test this on macos/windows currently.